### PR TITLE
GTC: GridTools C++ codegen IR + generator

### PR DESCRIPTION
**Pull request by havogt**
## Description

Describe the content of the PR and links to related issues, bugs of features.

## Requirements

Before submitting this PR, please make sure:

- [ ] The code builds cleanly without new errors or warnings
- [ ] The code passes all the existing tests
- [ ] If this PR adds a new feature, new tests have been added to test these
new features
- [ ] All relevant documentation has been updated or added

Additionally, if this PR contains code authored by new contributors:

- [ ] All the authors are covered by a valid contributor assignment agreement,
signed by the employer if needed, provided to ETH Zurich
- [ ] The names of all the new contributors have been added to an updated
version of the AUTHORS.rst file included in the PR

### CODE REVIEW

File: src/gt4py/backend/gtc_backend/defir_to_gtir.py [hunk 1]
```diff
@@ -17,22 +31,25 @@ class DefIRToGTIR(IRNodeVisitor):
     def apply(cls, root, **kwargs):
         return cls().visit(root)
 
-    GT4PY_ITERATIONORDER_TO_GTIR_LOOPORDER: ClassVar[Mapping[IterationOrder, common.LoopOrder]] = MappingProxyType(
+    GT4PY_ITERATIONORDER_TO_GTIR_LOOPORDER: ClassVar[
+        Mapping[IterationOrder, common.LoopOrder]
+    ] = MappingProxyType(
```
Comment (DropD) [inline src/gt4py/backend/gtc_backend/defir_to_gtir.py]: I wonder if there is a reason this is not a simple `Dict`

Comment (DropD) [reply src/gt4py/backend/gtc_backend/defir_to_gtir.py]: Does it just come from observing that `Enum`s use MappingProxyType? The way I have seen these X_TO_Y mappings used, a `Dict` would be fine too.

Comment (egparedes) [reply src/gt4py/backend/gtc_backend/defir_to_gtir.py]: Yes, there is nothing special about these mappings and they could be also dicts. I think MappingProxyTypes were used here just to protect new Python users to make weird bugs by accidentally modifying the mappings.

**Comment by havogt** [comment]
replaced by #272

**Pull request closed without merging**